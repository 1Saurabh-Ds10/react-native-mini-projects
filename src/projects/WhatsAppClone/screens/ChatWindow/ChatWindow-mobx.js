import React, { Component } from 'react';

import {
  View,
  ImageBackground,
  KeyboardAvoidingView,
  Platform,
  Text,
  StyleSheet,
  Button,
  FlatList
} from 'react-native';
import { inject, observer } from 'mobx-react';

import Message from '../../components/Message/Message';
import Compose from '../../components/Compose/Compose';


import styles from './styles';


@inject("messageStore")
@observer
class ChatWindow extends Component {
  state = {
    messages: []
  };
  keyboardVerticalOffset = Platform.OS === 'ios' ? 60 : 0;

  static navigationOptions = ({ navigation }) => ({
    title: `Chat with ${navigation.state.params.name}`
  });

  componentDidMount(){
    this.props.messageStore.subscribeToGetMessagesFromServer();
}

componentWillUnmount(){
  this.props.messageStore.unSubscribeToGetMessagesFromServer();
}

  onBackBtnPress = () => {
    this.props.navigation.goBack();
  };

  render() {
    return (
      <View style={styles.container}>
        <ImageBackground style={styles.container} source={require('../../assets/images/background.png')}>
          <KeyboardAvoidingView
            behavior="padding"
            keyboardVerticalOffset={this.keyboardVerticalOffset}
            style={styles.container}
          >
            <FlatList
              style={styles.container}
              data={this.props.messageStore.messages}
              renderItem={Message}
              keyExtractor={(__item, index) => `message-${index}`}
            />
            <Compose submit={ this.props.messageStore.postMessageToServer } />
          </KeyboardAvoidingView>
        </ImageBackground>
        <Button onPress={this.onBackBtnPress} title="Navigate to Home Screen" />
      </View>
    );
  }
}


export default ChatWindow;
