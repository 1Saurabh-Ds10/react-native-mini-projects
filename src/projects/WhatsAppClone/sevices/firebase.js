import * as firebase from 'firebase';
import {
  DATABASE_API_KEY,
  DATABASE_AUTH_DOMAIN,
  DATABASE_URL,
  DATABASE_PROJECT_ID,
  DATABASE_STORAGE_BUCKET,
  DATABASE_MESSAGING_SENDER_ID
} from 'react-native-dotenv';


export const initialize = () =>
  firebase.initializeApp({
    apiKey: DATABASE_API_KEY,
    authDomain: DATABASE_AUTH_DOMAIN,
    databaseURL: DATABASE_URL,
    projectId: DATABASE_PROJECT_ID,
    storageBucket: DATABASE_STORAGE_BUCKET,
    messagingSenderId: DATABASE_MESSAGING_SENDER_ID
  });

export const setListener = (endpoint, updaterFn) => {
  firebase
    .database()
    .ref(endpoint)
    .on('value', updaterFn);
  return () =>
    firebase
      .database()
      .ref(endpoint)
      .off();
};

export const pushData = (endpoint, data) => {
  return firebase
    .database()
    .ref(endpoint)
    .push(data);
};
