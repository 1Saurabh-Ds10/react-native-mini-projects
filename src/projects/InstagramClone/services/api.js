import { signup, login, initialize } from './firebase';
export { login, signup };
export const initApi = () => initialize();
export const getMessages = updaterFn => setListener('messages', updaterFn);

export const postMessage = message => {
  if (Boolean(message)) {
    return pushData('messages', {
      incoming: true,
      message
    });
  }
};
