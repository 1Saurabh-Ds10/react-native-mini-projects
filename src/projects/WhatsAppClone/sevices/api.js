import { setListener, pushData, initialize } from './firebase';
export const mockMessages = [
  {
    incoming: true,
    message: 'Hi Max'
  },
  {
    incoming: false,
    message: 'Hi, John'
  },
  {
    incoming: true,
    message: 'When we will learn real data fetching'
  },
  {
    incoming: false,
    message: `First we need to learn debugging and testing, then we will learn styling and animations and then we will learn real data fetching. Let's use this mock data for now ok?`
  }
];

export const initApi = () => initialize();

export const getMessages = updaterFn => setListener('messages', updaterFn);

export const postMessage = message => {
  if (Boolean(message)) {
    return pushData('messages', {
      incoming: true,
      message
    });
  }
};

export const getMockData = () => new Promise(resolve => setTimeout(() => resolve(mockMessages), 1000));
