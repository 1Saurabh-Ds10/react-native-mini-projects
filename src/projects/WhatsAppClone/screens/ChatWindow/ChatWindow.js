import React, { Component } from 'react';
import { getMockData } from '../../sevices/api';
import { View, ImageBackground, Text, StyleSheet, Button, FlatList } from 'react-native';
import styles from './styles';

class ChatWindow extends Component {
  state = {
    messages: []
  };
  static navigationOptions = ({ navigation }) => ({
    title: `Chat with ${navigation.state.params.name}`
  });

  componentDidMount() {
    getMockData().then(messages => {
      this.setState({
        messages
      });
    });
  }

  onBackBtnPress = () => {
    this.props.navigation.goBack();
  };

  getMessageRow = item => {
    return (
      <View style={[styles.listItem, item.incoming ? styles.incomingMessage : styles.outgoingMessage]}>
        <Text>{item.message}</Text>
      </View>
    );
  };

  render() {
    return (
      <View style={styles.container}>
        
        <ImageBackground style={styles.container} source={require('../../assets/images/background.png')}>
          <FlatList
            data={this.state.messages}
            renderItem={({ item }) => this.getMessageRow(item)}
            keyExtractor={(__item, index) => `message-${index}`}
          />
        </ImageBackground>
        <Button onPress={this.onBackBtnPress} title="Navigate to Home Screen" />
      </View>
    );
  }
}

export default ChatWindow;
