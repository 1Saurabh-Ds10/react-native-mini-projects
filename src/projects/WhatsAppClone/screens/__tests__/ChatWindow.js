import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ChatWindow from '../ChatWindow/ChatWindow';
import { getMockDataSpy } from '../../sevices/api';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

jest.mock('../../sevices/api');

Enzyme.configure({ adapter: new Adapter() });

describe('ChatWindow', () => {
  it('Chat Screen renders correctly', () => {
    const tree = renderer.create(<ChatWindow />);
    expect(tree).toMatchSnapshot();
  });

  it('ChatWindow has proper navigation options', () => {
    expect(
      ChatWindow.navigationOptions({
        navigation: {
          state: {
            params: {
              name: 'John'
            }
          }
        }
      })
    ).toMatchSnapshot();
  });

  it('goBack on press', () => {
    const goBack = jest.fn();
    const ChatWindow = shallow(<ChatWindow navigation={{ goBack }} />);
    ChatWindow.find('Button').simulate('press');
    expect(goBack).toBeCalledWith();
  });

  it('fetches data when component is rendered', () => {
    const ChatWindow = renderer.create(<ChatWindow />);
    expect(getMockDataSpy).toBeCalled();
  });
});
