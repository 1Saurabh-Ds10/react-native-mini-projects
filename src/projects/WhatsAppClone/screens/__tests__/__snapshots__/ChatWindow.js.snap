// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatWindow Chat Screen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Chat with John
    </Text>
  </View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Navigate to Home Screen
      </Text>
    </View>
  </View>
</View>
`;

exports[`ChatWindow ChatWindow has proper navigation options 1`] = `
Object {
  "title": "Chat with John",
}
`;
